packages/interactivity/src/directives.tsx

```diff
@@ -354,19 +354,24 @@ export default () => {
 				.forEach( ( entry ) => {
 					const className = entry.suffix;
 					const result = evaluate( entry );
-					const currentClass = element.props.class || '';
-					const classFinder = new RegExp(
-						`(^|\\s)${ className }(\\s|$)`,
-						'g'
-					);
```

Comment:
Independent of the other changes, this is likely a good change. Building a regex effectively from an input string like was done here is risky.

Reply:
This change is _necessary_ here because there are number of special characters that would need special processing in the Regexp like `[]`.